pkg/proc/target_exec.go

```diff
@@ -191,17 +223,56 @@ func (grp *TargetGroup) Continue() error {
 			if curbp.Breakpoint.WatchType != 0 {
 				dbp.StopReason = StopWatchpoint
 			}
-			return conditionErrors(threads)
+			return conditionErrors(grp)
 		default:
 			// not a manual stop, not on runtime.Breakpoint, not on a breakpoint, just repeat
 		}
 		if callInjectionDone {
 			// a call injection was finished, don't let a breakpoint with a failed
 			// condition or a step breakpoint shadow this.
-			dbp.StopReason = StopCallReturned
-			return conditionErrors(threads)
+			return conditionErrors(grp)
+		}
+	}
+}
+
+func (grp *TargetGroup) finishManualStop() {
+	for _, dbp := range grp.targets {
+		if isvalid, _ := dbp.Valid(); !isvalid {
+			continue
+		}
+		dbp.StopReason = StopManual
+		dbp.clearHardcodedBreakpoints()
+		if grp.KeepSteppingBreakpoints&HaltKeepsSteppingBreakpoints == 0 {
+			dbp.ClearSteppingBreakpoints()
+		}
+	}
+}
+
+// fixCurrentThreads switches traptgt to trapthread, then for each target in
+// the group if its current thread exists it refreshes the current
+// goroutine, otherwise it switches it to a randomly selected thread.
+func (grp *TargetGroup) fixCurrentThreads(traptgt *Target, trapthread Thread) error {
```

Comment:
nit: maybe `setCurrentThreads` instead. The `fix` prefix implies something was done incorrectly.